### PR TITLE
workflows: html raw marc views fixed

### PR DESCRIPTION
**Pull request by GiorgosPa**
- Fixes HTML, RAW and MARC views in details page.

### CODE REVIEW

File: invenio/modules/workflows/views/holdingpen.py [hunk 1]
```diff
@@ -239,16 +239,25 @@ def entry_data_preview(objectid, of):
     from pprint import pformat
 
     bwobject = BibWorkflowObject.query.get(int(objectid))
-
+    # import ipdb
```
Comment (jalavik) [inline invenio/modules/workflows/views/holdingpen.py]: This can be removed.

File: invenio/modules/workflows/views/holdingpen.py [hunk 2]
```diff
@@ -239,16 +239,25 @@ def entry_data_preview(objectid, of):
...
         flash("No object found for %s" % (objectid,))
         return jsonify(data={})
-
-    formatted_data = bwobject.get_formatted_data()
+    formatted_data = bwobject.get_formatted_data(of)
     if isinstance(formatted_data, dict):
         formatted_data = pformat(formatted_data)
-    if of and of in ("xm", "xml", "marcxml"):
+    if of == "xm":
+        formatted_data = formatted_data.replace('\n', '')
```
Comment (jalavik) [inline invenio/modules/workflows/views/holdingpen.py]: It is cleaner if we do the prettify-ing at a lower level. For example in `get_formatted_data` (which from it's name should already have it formatted and nice).

File: invenio/modules/workflows/views/holdingpen.py [hunk 3]
```diff
@@ -239,16 +239,25 @@ def entry_data_preview(objectid, of):
...
+        formatted_data = formatted_data.replace('<record>', '\n<record>')
+        formatted_data = formatted_data.replace('</record>', '\n</record>')
+        formatted_data = formatted_data.replace('<controlfield', '\n\t<controlfield')
+        formatted_data = formatted_data.replace('<datafield', '\n\t<datafield')
+        formatted_data = formatted_data.replace('<subfield', '\n\t\t<subfield')
+        formatted_data = formatted_data.replace('</datafield>', '\n\t</datafield>')
         data = Markup.escape(formatted_data)
+    elif of == 'xo':
+        data = Markup.escape(formatted_data[0])
```
Comment (jalavik) [inline invenio/modules/workflows/views/holdingpen.py:275]: What does the `xo` format return usually? Just to understand why it is indexed.

Comment (GiorgosPa) [inline invenio/modules/workflows/views/holdingpen.py:275]: It returns a human readable xml with tags like <title>, <abstract> etc

**Comment by GiorgosPa** [comment]
Pull request updated.

**Comment by jalavik** [comment]
Thanks. Cherry picked as 409c24a

**Pull request closed without merging**